src/pf-popover/pf-popover.component.js

```diff
@@ -0,0 +1,421 @@
+    closeButton.innerHTML = `<button type="button" class="close"><span class="pficon pficon-close"></span></button>`;
+
+    if (this._popoverTitle === '' && !this._dismissible) {
+      popovertitle.parentNode.removeChild(popovertitle);
+    } else {
+      //set popover title
+      popovertitle.innerHTML = this._popoverTitle;
+
+      if (this._dismissible) {
+        popovertitle.appendChild(closeButton.content);
+      }
+    }
+    //set popover content
+    popoverInner.innerHTML = this.content;
+
+    //append to the container
+    this._container.appendChild(clone);
+
+    //set reference to appended node
+    let popovers = this._container.querySelectorAll('.popover');
+    this.popover = popovers[popovers.length - 1];
+    this.popover.style.display = 'block';
+    this.popover.setAttribute('class', `popover ${this._placement} ${this._animation}`);
+  }
+
+  /**
+   * update the placement of popover
+   */
+  _updatePlacement() {
+    return this._placement === 'top' ? 'bottom' : // top
+      this._placement === 'bottom' ? 'top' : // bottom
+        this._placement === 'left' ? 'right' : // left
+          this._placement === 'right' ? 'left' : this._placement; // right
+  }
+
+  /**
+   * Styles the popover based on placement attribute
+   * @private
+   */
+  _stylePopover() {
+    const rect = this._target.getBoundingClientRect(); //popover real dimensions
+
+    const // link rect | window vertical and horizontal scroll
+      scroll = pfUtil.getScroll();
+
+    const //link real dimensions
+      linkDimensions = { w: rect.right - rect.left, h: rect.bottom - rect.top };
+
+    const popoverDimensions = { w: this.popover.offsetWidth, h: this.popover.offsetHeight };
+
+    //apply styling
+    switch (this._placement) {
+      case 'top': //TOP
+        this.popover.style.top = `${rect.top + scroll.y - popoverDimensions.h}px`;
+        this.popover.style.left = `${rect.left + scroll.x - popoverDimensions.w / 2 + linkDimensions.w / 2}px`;
+        break;
+
+      case 'bottom': //BOTTOM
+        this.popover.style.top = `${rect.top + scroll.y + linkDimensions.h}px`;
+        this.popover.style.left = `${rect.left + scroll.x - popoverDimensions.w / 2 + linkDimensions.w / 2}px`;
+        break;
+
+      case 'left': //LEFT
+        this.popover.style.top = `${rect.top + scroll.y - popoverDimensions.h / 2 + linkDimensions.h / 2}px`;
+        this.popover.style.left = `${rect.left + scroll.x - popoverDimensions.w}px`;
+        break;
+
+      case 'right': //RIGHT
+        this.popover.style.top = `${rect.top + scroll.y - popoverDimensions.h / 2 + linkDimensions.h / 2}px`;
+        this.popover.style.left = `${rect.left + scroll.x + linkDimensions.w}px`;
+        break;
+    }
+  }
+
+  /**
+   * check the placement of popover
+   */
+  _checkPlacement() {
+    if (!pfUtil.isElementInViewport(this.popover)) {
+      this._placement = this._updatePlacement();
+      this._createPopover();
+      this._stylePopover();
```

Comment:
Continuation... see comment above, related to causing multiple DOM nodes to be appended